tests/response/test_serialization.py

```diff
@@ -53,7 +53,7 @@ def test_response_serialization_structured_types(content: Any, response_type: An
     if isinstance(value, dict) and "enum" in value:
         assert content.__class__(**value)["enum"] == content["enum"].value
     elif isinstance(value, dict) and "secret" in value:
-        assert content.__class__(**value)["secret"] == content["secret"].get_secret_value()
+        assert content.__class__(**value)["secret"] == str(content["secret"])
```

Comment:
Because this PR changes the behaviour of the secret fields to match how Pydantic handles them.

My reasoning for this was that since Pydantic cloaks the values when serializing, so should we, as this is what people familiar with Pydantic expect, and is also the behaviour shown in the Pydantic docs.